modules/navier_stokes/include/fv/kernels/NSFVAdvectionBase.h

```diff
@@ -70,16 +71,18 @@ class NSFVAdvectionBase
   /**
    * Clear the RC 'a' coefficient cache
    */
-  void clearRCCoeffs() { _rc_a_coeffs[_nsfv_tid].clear(); }
+  void clearRCCoeffs();
 
 private:
+  const MooseApp & _nsfv_app;
   const SubProblem & _nsfv_subproblem;
-
   const THREAD_ID _nsfv_tid;
 
   /// A map from elements to the 'a' coefficients used in the Rhie-Chow interpolation. The size of
-  /// the vector is equal to the number of threads in the simulation
-  static std::vector<std::unordered_map<const Elem *, ADReal>> _rc_a_coeffs;
+  /// the vector is equal to the number of threads in the simulation. We maintain a map from
+  /// MooseApp pointer to RC coefficients in order to support MultiApp simulations
+  static std::unordered_map<const MooseApp *, std::vector<std::unordered_map<const Elem *, ADReal>>>
+      _rc_a_coeffs;
```

Comment:
why an unordered map for the inner map? Ordered would be faster. Maybe by Elem ID?

Reply:
or is it easier for refinement or something like that?

Reply:
When talking about faster, are you referring to insertions or lookup? `unordered_map` is amortized O(1) vs. `map` O(n log N) for lookups. `unordered_map` is certainly generally heavier in memory

Reply:
true true. Ordering by ID could even use a vector.  
But I guess vectors wouldn't be great because of block restrictions, so many elem IDs may not require rc_a_coeffs.